File: win/directx/sdk-9.0/Samples/VB.Net/DirectPlay/Tutorials/Tut05_Send/ConnectDialog.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports Microsoft.DirectX.DirectPlay

Namespace Tut05_Send

    '/ <summary>
    '/ Summary description for ConnectDialog.
    '/ </summary>
    Public Class ConnectDialog
        Inherits System.Windows.Forms.Form
        Private App As SendApp = Nothing ' Application instance
        Private m_SelectedHost As HostInfo = Nothing ' Host selected for connection
        Private label1 As System.Windows.Forms.Label
        Private label2 As System.Windows.Forms.Label
        Private pictureBox1 As System.Windows.Forms.PictureBox
        Private label3 As System.Windows.Forms.Label
        Private label4 As System.Windows.Forms.Label
        Private Shadows WithEvents cancelButton As System.Windows.Forms.Button
        Private WithEvents ConnectButton As System.Windows.Forms.Button
        Private WithEvents SearchButton As System.Windows.Forms.Button
        Private SearchAddressTextBox As System.Windows.Forms.TextBox
        Private WithEvents DetectedSessionsListBox As System.Windows.Forms.ListBox

        ' Properties

        Public ReadOnly Property SelectedHost() As HostInfo
            Get
                Return m_SelectedHost
            End Get
        End Property '/ <summary>
        '/ Required designer variable.
        '/ </summary>

        ' Property: Remote port on which to connect
        Public Property RemotePort() As Integer
            Get
                Dim retValue As Integer

                Try
                    retValue = Integer.Parse(RemotePortTextBox.Text)
                Catch ex As Exception
                End Try

                Return retValue
            End Get

            Set(ByVal Value As Integer)
                RemotePortTextBox.Text = Value.ToString()
            End Set
        End Property

        Private components As System.ComponentModel.Container = Nothing


        Public Sub New(ByVal app As SendApp)
            Me.App = app

            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()

            '
            ' TODO: Add any constructor code after InitializeComponent call
            '
            SearchAddressTextBox.Text = "localhost"
            DetectedSessionsListBox.Enabled = False
            ConnectButton.Enabled = False
            DetectedSessionsListBox.Items.Clear()
            DetectedSessionsListBox.Items.Add("Click ""Search"" to find hosts.")
        End Sub 'New

        '/ <summary>
        '/ Clean up any resources being used.
        '/ </summary>
        Protected Overloads Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub 'Dispose

        '/ <summary>
        '/ Required method for Designer support - do not modify
        '/ the contents of this method with the code editor.
        '/ </summary>
        Friend WithEvents RemotePortTextBox As System.Windows.Forms.TextBox
        Friend WithEvents remotePortLabel As System.Windows.Forms.Label
        Private Sub InitializeComponent()
            Me.label1 = New System.Windows.Forms.Label()
            Me.label2 = New System.Windows.Forms.Label()
            Me.pictureBox1 = New System.Windows.Forms.PictureBox()
            Me.label3 = New System.Windows.Forms.Label()
            Me.label4 = New System.Windows.Forms.Label()
            Me.cancelButton = New System.Windows.Forms.Button()
            Me.ConnectButton = New System.Windows.Forms.Button()
            Me.SearchAddressTextBox = New System.Windows.Forms.TextBox()
            Me.SearchButton = New System.Windows.Forms.Button()
            Me.DetectedSessionsListBox = New System.Windows.Forms.ListBox()
            Me.RemotePortTextBox = New System.Windows.Forms.TextBox()
            Me.remotePortLabel = New System.Windows.Forms.Label()
            Me.SuspendLayout()
            '
            'label1
            '
            Me.label1.BackColor = System.Drawing.SystemColors.Info
            Me.label1.Location = New System.Drawing.Point(24, 16)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(312, 16)
            Me.label1.TabIndex = 0
            Me.label1.Text = "Find hosts by using the search field below. Select a detected"
            '
            'label2
            '
            Me.label2.BackColor = System.Drawing.SystemColors.Info
            Me.label2.Location = New System.Drawing.Point(24, 32)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(312, 16)
            Me.label2.TabIndex = 1
            Me.label2.Text = "host and click ""Connect"" to join the session."
            '
            'pictureBox1
            '
            Me.pictureBox1.BackColor = System.Drawing.SystemColors.Info
            Me.pictureBox1.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
            Me.pictureBox1.Location = New System.Drawing.Point(8, 8)
            Me.pictureBox1.Name = "pictureBox1"
            Me.pictureBox1.Size = New System.Drawing.Size(336, 48)
            Me.pictureBox1.TabIndex = 2
            Me.pictureBox1.TabStop = False
            '
            'label3
            '
            Me.label3.Location = New System.Drawing.Point(40, 72)
            Me.label3.Name = "label3"
            Me.label3.Size = New System.Drawing.Size(100, 16)
            Me.label3.TabIndex = 3
            Me.label3.Text = "Search Address:"
            '
            'label4
            '
            Me.label4.Location = New System.Drawing.Point(40, 120)
            Me.label4.Name = "label4"
            Me.label4.Size = New System.Drawing.Size(120, 16)
            Me.label4.TabIndex = 4
            Me.label4.Text = "Detected Sessions:"
            '
            'cancelButton
            '
            Me.cancelButton.DialogResult = System.Windows.Forms.DialogResult.Cancel
            Me.cancelButton.Location = New System.Drawing.Point(192, 216)
            Me.cancelButton.Name = "cancelButton"
            Me.cancelButton.Size = New System.Drawing.Size(72, 24)
            Me.cancelButton.TabIndex = 5
            Me.cancelButton.Text = "&Cancel"
            '
            'ConnectButton
            '
            Me.ConnectButton.Location = New System.Drawing.Point(272, 216)
            Me.ConnectButton.Name = "ConnectButton"
            Me.ConnectButton.Size = New System.Drawing.Size(72, 24)
            Me.ConnectButton.TabIndex = 6
            Me.ConnectButton.Text = "C&onnect"
            '
            'SearchAddressTextBox
            '
            Me.SearchAddressTextBox.AutoSize = False
            Me.SearchAddressTextBox.Location = New System.Drawing.Point(40, 88)
            Me.SearchAddressTextBox.Name = "SearchAddressTextBox"
            Me.SearchAddressTextBox.Size = New System.Drawing.Size(136, 24)
            Me.SearchAddressTextBox.TabIndex = 7
            Me.SearchAddressTextBox.Text = "localhost"
            '
            'SearchButton
            '
            Me.SearchButton.Location = New System.Drawing.Point(232, 88)
            Me.SearchButton.Name = "SearchButton"
            Me.SearchButton.Size = New System.Drawing.Size(72, 24)
            Me.SearchButton.TabIndex = 8
            Me.SearchButton.Text = "&Search"
            '
            'DetectedSessionsListBox
            '
            Me.DetectedSessionsListBox.Location = New System.Drawing.Point(40, 136)
            Me.DetectedSessionsListBox.Name = "DetectedSessionsListBox"
            Me.DetectedSessionsListBox.Size = New System.Drawing.Size(264, 56)
            Me.DetectedSessionsListBox.TabIndex = 9
            '
            'RemotePortTextBox
            '
            Me.RemotePortTextBox.AutoSize = False
            Me.RemotePortTextBox.Location = New System.Drawing.Point(182, 88)
            Me.RemotePortTextBox.Name = "RemotePortTextBox"
            Me.RemotePortTextBox.Size = New System.Drawing.Size(40, 24)
            Me.RemotePortTextBox.TabIndex = 20
            Me.RemotePortTextBox.Text = ""
            '
            'remotePortLabel
            '
            Me.remotePortLabel.Location = New System.Drawing.Point(182, 72)
            Me.remotePortLabel.Name = "remotePortLabel"
            Me.remotePortLabel.Size = New System.Drawing.Size(40, 16)
            Me.remotePortLabel.TabIndex = 19
            Me.remotePortLabel.Text = "Port:"
            '
            'ConnectDialog
            '
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(352, 246)
            Me.ControlBox = False
            Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.RemotePortTextBox, Me.remotePortLabel, Me.DetectedSessionsListBox, Me.SearchButton, Me.SearchAddressTextBox, Me.ConnectButton, Me.cancelButton, Me.label4, Me.label3, Me.label2, Me.label1, Me.pictureBox1})
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "ConnectDialog"
            Me.ShowInTaskbar = False
            Me.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
            Me.Text = "Connect to Session"
            Me.ResumeLayout(False)

        End Sub 'InitializeComponent

        '/ <summary>
        '/ Handler for Search button click
        '/ </summary>
        Private Sub SearchButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles SearchButton.Click
            SearchButton.Enabled = False
            Me.Cursor = Cursors.WaitCursor

            ' Clear the current UI list
            DetectedSessionsListBox.Enabled = False
            DetectedSessionsListBox.Items.Clear()

            ' Have the application perform the enumeration
            App.EnumerateSessions(SearchAddressTextBox.Text, RemotePort)

            ' Add detected items to thelist
            Dim host As HostInfo
            For Each host In App.FoundSessions
                DetectedSessionsListBox.Items.Add(host)
                DetectedSessionsListBox.Enabled = True
            Next host

            ' Give default message for no hosts
            If App.FoundSessions.Count = 0 Then
                DetectedSessionsListBox.Items.Add("No hosts found.")
            End If
            Me.Cursor = Cursors.Default
            SearchButton.Enabled = True
        End Sub 'SearchButton_Click

        '/ <summary>
        '/ Handler for Connect button click
        '/ </summary>
        Private Sub ConnectButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles ConnectButton.Click
            ' Save the current settings
            m_SelectedHost = CType(DetectedSessionsListBox.SelectedItem, HostInfo)
            DialogResult = DialogResult.OK
        End Sub 'ConnectButton_Click

        '/ <summary>
        '/ Handler for Cancel button click
        '/ </summary>
        Private Sub cancelButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles cancelButton.Click
            ' Discard the settings
            DialogResult = DialogResult.Cancel
        End Sub 'cancelButton_Click

        '/ <summary>
        '/ Handler for Detected Sessions selection change
        '/ </summary>
        Private Sub DetectedSessionsListBox_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles DetectedSessionsListBox.SelectedIndexChanged
            ConnectButton.Enabled = DetectedSessionsListBox.SelectedIndex <> -1
        End Sub 'DetectedSessionsListBox_SelectedIndexChanged
    End Class 'ConnectDialog

End Namespace

File: win/directx/sdk-9.0/Samples/C#/DirectPlay/Tutorials/Tut10_ThreadPool/ConnectDialog.cs
//----------------------------------------------------------------------------
// File: ConnectDialog.cs
//
// Copyright (c) Microsoft Corp. All rights reserved.
//-----------------------------------------------------------------------------
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using Microsoft.DirectX.DirectPlay;

namespace Tut10_ThreadPool
{
    /// <summary>
    /// Summary description for ConnectDialog.
    /// </summary>
    public class ConnectDialog : System.Windows.Forms.Form
    {
        private ThreadPoolApp App = null;                      // Application instance
        private HostInfo         mySelectedHost = null;           // Host selected for connection
          
        private System.Windows.Forms.Label label1;
        private System.Windows.Forms.Label label2;
        private System.Windows.Forms.PictureBox pictureBox1;
        private System.Windows.Forms.Label label3;
        private System.Windows.Forms.Label label4;
        private System.Windows.Forms.Button cancelButton;
        private System.Windows.Forms.Button ConnectButton;
        private System.Windows.Forms.Button SearchButton;
        private System.Windows.Forms.TextBox SearchAddressTextBox;
        private System.Windows.Forms.ListBox DetectedSessionsListBox;
        private System.Windows.Forms.Label remotePortLabel;
        public System.Windows.Forms.TextBox RemotePortTextBox;
		
        // Properties
        public  HostInfo SelectedHost { get { return mySelectedHost; } }

        /// <summary>
        /// Property: remote port on which to connect
        /// </summary>
        public int RemotePort
        {
            get
            { 
                int retValue = 0;

                try
                {
                    retValue = int.Parse(RemotePortTextBox.Text); 
                }
                catch (Exception) {}
                
                return retValue;
            }
            set
            { RemotePortTextBox.Text = value.ToString(); }
        }

        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components = null;

        public ConnectDialog(ThreadPoolApp app)
        {
            App = app;

            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
            SearchAddressTextBox.Text = "localhost";
            DetectedSessionsListBox.Enabled = false;
            ConnectButton.Enabled = false;
            DetectedSessionsListBox.Items.Clear();
            DetectedSessionsListBox.Items.Add("Click \"Search\" to find hosts.");
        }

        

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

		#region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.pictureBox1 = new System.Windows.Forms.PictureBox();
            this.label3 = new System.Windows.Forms.Label();
            this.label4 = new System.Windows.Forms.Label();
            this.cancelButton = new System.Windows.Forms.Button();
            this.ConnectButton = new System.Windows.Forms.Button();
            this.SearchAddressTextBox = new System.Windows.Forms.TextBox();
            this.SearchButton = new System.Windows.Forms.Button();
            this.DetectedSessionsListBox = new System.Windows.Forms.ListBox();
            this.remotePortLabel = new System.Windows.Forms.Label();
            this.RemotePortTextBox = new System.Windows.Forms.TextBox();
            this.SuspendLayout();
            // 
            // label1
            // 
            this.label1.BackColor = System.Drawing.SystemColors.Info;
            this.label1.Location = new System.Drawing.Point(24, 16);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(312, 16);
            this.label1.TabIndex = 0;
            this.label1.Text = "Find hosts by using the search field below. Select a detected";
            // 
            // label2
            // 
            this.label2.BackColor = System.Drawing.SystemColors.Info;
            this.label2.Location = new System.Drawing.Point(24, 32);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(312, 16);
            this.label2.TabIndex = 1;
            this.label2.Text = "host and click \"Connect\" to join the session.";
            // 
            // pictureBox1
            // 
            this.pictureBox1.BackColor = System.Drawing.SystemColors.Info;
            this.pictureBox1.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
            this.pictureBox1.Location = new System.Drawing.Point(8, 8);
            this.pictureBox1.Name = "pictureBox1";
            this.pictureBox1.Size = new System.Drawing.Size(336, 48);
            this.pictureBox1.TabIndex = 2;
            this.pictureBox1.TabStop = false;
            // 
            // label3
            // 
            this.label3.Location = new System.Drawing.Point(40, 72);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(100, 16);
            this.label3.TabIndex = 3;
            this.label3.Text = "Search Address:";
            // 
            // label4
            // 
            this.label4.Location = new System.Drawing.Point(40, 120);
            this.label4.Name = "label4";
            this.label4.Size = new System.Drawing.Size(120, 16);
            this.label4.TabIndex = 4;
            this.label4.Text = "Detected Sessions:";
            // 
            // cancelButton
            // 
            this.cancelButton.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.cancelButton.Location = new System.Drawing.Point(192, 216);
            this.cancelButton.Name = "cancelButton";
            this.cancelButton.Size = new System.Drawing.Size(72, 24);
            this.cancelButton.TabIndex = 5;
            this.cancelButton.Text = "&Cancel";
            this.cancelButton.Click += new System.EventHandler(this.cancelButton_Click);
            // 
            // ConnectButton
            // 
            this.ConnectButton.Location = new System.Drawing.Point(272, 216);
            this.ConnectButton.Name = "ConnectButton";
            this.ConnectButton.Size = new System.Drawing.Size(72, 24);
            this.ConnectButton.TabIndex = 6;
            this.ConnectButton.Text = "C&onnect";
            this.ConnectButton.Click += new System.EventHandler(this.ConnectButton_Click);
            // 
            // SearchAddressTextBox
            // 
            this.SearchAddressTextBox.AutoSize = false;
            this.SearchAddressTextBox.Location = new System.Drawing.Point(40, 88);
            this.SearchAddressTextBox.Name = "SearchAddressTextBox";
            this.SearchAddressTextBox.Size = new System.Drawing.Size(136, 24);
            this.SearchAddressTextBox.TabIndex = 7;
            this.SearchAddressTextBox.Text = "localhost";
            // 
            // SearchButton
            // 
            this.SearchButton.Location = new System.Drawing.Point(232, 88);
            this.SearchButton.Name = "SearchButton";
            this.SearchButton.Size = new System.Drawing.Size(72, 24);
            this.SearchButton.TabIndex = 8;
            this.SearchButton.Text = "&Search";
            this.SearchButton.Click += new System.EventHandler(this.SearchButton_Click);
            // 
            // DetectedSessionsListBox
            // 
            this.DetectedSessionsListBox.Location = new System.Drawing.Point(40, 136);
            this.DetectedSessionsListBox.Name = "DetectedSessionsListBox";
            this.DetectedSessionsListBox.Size = new System.Drawing.Size(264, 56);
            this.DetectedSessionsListBox.TabIndex = 9;
            this.DetectedSessionsListBox.SelectedIndexChanged += new System.EventHandler(this.DetectedSessionsListBox_SelectedIndexChanged);
            // 
            // remotePortLabel
            // 
            this.remotePortLabel.Location = new System.Drawing.Point(181, 72);
            this.remotePortLabel.Name = "remotePortLabel";
            this.remotePortLabel.Size = new System.Drawing.Size(32, 16);
            this.remotePortLabel.TabIndex = 19;
            this.remotePortLabel.Text = "Port:";
            // 
            // RemotePortTextBox
            // 
            this.RemotePortTextBox.AutoSize = false;
            this.RemotePortTextBox.Location = new System.Drawing.Point(181, 88);
            this.RemotePortTextBox.Name = "RemotePortTextBox";
            this.RemotePortTextBox.Size = new System.Drawing.Size(40, 24);
            this.RemotePortTextBox.TabIndex = 20;
            this.RemotePortTextBox.Text = "";
            // 
            // ConnectDialog
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.CancelButton = this.cancelButton;
            this.ClientSize = new System.Drawing.Size(352, 246);
            this.ControlBox = false;
            this.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                          this.remotePortLabel,
                                                                          this.RemotePortTextBox,
                                                                          this.DetectedSessionsListBox,
                                                                          this.SearchButton,
                                                                          this.SearchAddressTextBox,
                                                                          this.ConnectButton,
                                                                          this.cancelButton,
                                                                          this.label4,
                                                                          this.label3,
                                                                          this.label2,
                                                                          this.label1,
                                                                          this.pictureBox1});
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "ConnectDialog";
            this.ShowInTaskbar = false;
            this.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide;
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Connect to Session";
            this.ResumeLayout(false);

        }
		#endregion

        /// <summary>
        /// Handler for Search button click
        /// </summary>
        private void SearchButton_Click(object sender, System.EventArgs e)
        {
            SearchButton.Enabled = false;
            this.Cursor = Cursors.WaitCursor;
            
            // Clear the current UI list
            DetectedSessionsListBox.Enabled = false;
            DetectedSessionsListBox.Items.Clear();

            // Have the application perform the enumeration
            App.EnumerateSessions(SearchAddressTextBox.Text, RemotePort);

            // Add detected items to thelist
            foreach (HostInfo host in App.FoundSessions)
            {
                DetectedSessionsListBox.Items.Add(host);
                DetectedSessionsListBox.Enabled = true;
            }

            // Give default message for no hosts
            if (App.FoundSessions.Count == 0)
                DetectedSessionsListBox.Items.Add("No hosts found.");

            this.Cursor = Cursors.Default;
            SearchButton.Enabled = true;
        }

        /// <summary>
        /// Handler for Connect button click
        /// </summary>
        private void ConnectButton_Click(object sender, System.EventArgs e)
        {
            // Save the current settings
            mySelectedHost = (HostInfo) DetectedSessionsListBox.SelectedItem;
            DialogResult = DialogResult.OK;
        }

        /// <summary>
        /// Handler for Cancel button click
        /// </summary>
        private void cancelButton_Click(object sender, System.EventArgs e)
        {
            // Discard the settings
            DialogResult = DialogResult.Cancel;
        }

        /// <summary>
        /// Handler for Detected Sessions selection change
        /// </summary>
        private void DetectedSessionsListBox_SelectedIndexChanged(object sender, System.EventArgs e)
        {
            ConnectButton.Enabled = (DetectedSessionsListBox.SelectedIndex != -1);
        }
    }
}
